modules/json/src/smithy4s/http/json/SchemaVisitorJCodec.scala

```diff
@@ -1094,7 +1096,12 @@ private[smithy4s] class SchemaVisitorJCodec(
       ): (Z, JsonWriter) => Unit = {
         val codec = apply(instance)
         val jLabel = jsonLabel(field)
-        if (jLabel.forall(JsonWriter.isNonEscapedAscii)) {
+        val hasDefault = instance.hints.has[Default]
+        val hasInternal = instance.hints.has[Internal]
+        if (hasInternal && hasDefault) { (_, _) =>
+          // we hide the field
+          ()
```

Comment:
as far as I understand, we're supposed to hide the field only if the default value would've been used. What if the following are all true:

- has internal
- has default
- _is_ not default (i.e. a different value is being serialized)

?

Reply:
> we're supposed to hide the field only if the default value would've been used
this sounds ridiculous but also true, should I ask for clarifications?

Reply:
probably

Reply:
Opened https://github.com/awslabs/smithy/issues/1565

Reply:
to quote Michael:

> Yes, a server should serialize fields annotated with internal (otherwise those members wouldn't be useful for anything other than requests). There are caveats mentioned around whether to serialize defaulted values marked as internal, but those are shoulds not musts.

Reply:
IMO we should either update the condition in question to

`if (hasInternal && hasDefault && <value>.isEmpty)`

or drop the PR altogether, thus not implementing a "should"

Reply:
At the moment, I concur with Jakub on this. Imho, "internal" should really have been an AWS specific trait, not exposed in the smithy prelude. 

@daddykotex can you formulate a compelling usecase for internal ? It may change my mind

Reply:
no, I can't

it's really just that I saw this and realized we did not support it and maybe user would have assumed it was supported. Maybe I can amend the documentation to add a note about internal.